heterarchy.litcoffee

```diff
@@ -63,7 +63,18 @@ behaves like a class that would be have such a hierarchy.
             class Result extends generate tail linearization
                 __mro__: linearization
                 constructor: reparent next, @, next::constructor
+                # FILL UP MISSING CLASS ATTRIBUTES
+                # copy class attributes from `next` to `this` that only exist on `next`
                 copyOwn next, @
+                # ADJUST CLASS METHOD (so the MRO is used)
+                # copy class methods from `next` to `this` that are implemented by `next`
+                # note: this is basically the same as copyOwn but ignoring `if not to.hasOwnProperty key`
+                # This is due to the fact that when inheriting all class attributes are copied to `this` by CoffeeScript.
+                for own key, value of next when value instanceof Function
+                    @[key] = partial(reparent, next, @)(value)
```

Comment:
Those two lines are equivalent.

I used `partial` to *curry* [1] the function passed to `copyOwn`, (`copyOwn` takes a function that *projects* the copied values, similar to `Array.map`). This is, we use `partial` to create a new function that binds `reparent`, `next`, and `@` to `reparent`, but leaves the last `value` argument free, thus creating a unary (with one argument) function out of a function with more arguments. These two expressions are equivalent:
```coffee
copyOwn next::, @::, partial reparent, next, @
copyOwn next::, @::, (v) -> reparent, next, @, v
```
i used `partial` because I was used to it from other languages tha support this pattern (it is also in Python `functools` btw.).

Of course, writing something like this...
```coffee
partial(reparent, next, @) value
```
...works but it is redundant, because it creates a new closure but then immediatelly applies it... we could just apply it to begin with using `reparent next, @, value` as you suggest. I think you added that line of code when inlining the `copyOwn` function, but forgot to also *inline* the partial call 😄 

[1] https://en.wikipedia.org/wiki/Currying